src/main/java/dev/joopie/jambot/music/command/PlayCommandHandler.java

```diff
@@ -36,17 +33,21 @@
 public class PlayCommandHandler extends ListenerAdapter implements CommandHandler {
     private static final String COMMAND_NAME = "play";
     private static final String COMMAND_OPTION_INPUT_NAME = "input";
-    private static final String SPOTIFY_URL = "spotify";
-    private static final Pattern URL_OR_ID_PATTERN = Pattern.compile("^(http(|s)://.*|[\\w\\-]{11})$");
-    public static final Pattern URL_PATTERN = Pattern.compile("^(http(|s)://.*)$");
-    public static final int SECONDS_OFFSET = 15;
+    private static final String SPOTIFY = "spotify";
     private static final String YOUTUBE_URL = "https://www.youtube.com/watch?v=";
+    private static final Pattern URL_OR_ID_PATTERN = Pattern.compile("^(http(|s)://.*|[\\w\\-]{11})$");
+    private static final Pattern URL_PATTERN = Pattern.compile("^(http(|s)://.*)$");
+    private static final Pattern YOU_TUBE_URL_PATTERN = Pattern.compile("^(https?)?(://)?(www\\.)?(m\\.)?((youtube\\.com)|(youtu\\.be))/");
+    private static final Pattern[] VIDEO_ID_PATTERNS = {
+            Pattern.compile("\\?vi?=([^&]*)"),
+            Pattern.compile("watch\\?.*v=([^&]*)"),
+            Pattern.compile("(?:embed|vi?)/([^/?]*)"),
+            Pattern.compile("^([A-Za-z0-9\\-]*)")
+    };
```

Comment:
I have added more pattern to make sure that all of the existing Youtube URLS are working. This took me a while. And yes. This is a copy paste from the internet because why should you make your own regex. ThisIsHell